src/lib/bm_integration/bcmp_cli.cpp

```diff
@@ -224,74 +224,46 @@ static BaseType_t cmd_bcmp_fn(char *writeBuffer, size_t writeBufferLen,
-            printf("Failed to send message config set\n");
-          } else {
-            printf("Succesfully sent config set msg\n");
+            printf("Failed to encode uint message...");
           }
           break;
         }
         case INT32: {
           int32_t val = strtol(value_str, NULL, 0);
           if (cbor_encode_int(&encoder, val) != CborNoError) {
-            break;
-          }
-          if (!bcmp_config_set(node_id, partition, key_str_str_len, key_str, buffer_size,
-                               cbor_buf, &err, NULL)) {
-            printf("Failed to send message config get\n");
-          } else {
-            printf("Succesfully sent config get msg\n");
+            printf("Failed to encode int message...");
           }
           break;
         }
         case FLOAT: {
-          float val;
-          if (!bStrtof(const_cast<char *>(value_str), &val)) {
-            printf("Invalid param\n");
-            break;
-          }
+          float val = strtof(const_cast<char *>(value_str), NULL);
           if (cbor_encode_float(&encoder, val) != CborNoError) {
-            break;
-          }
-          if (!bcmp_config_set(node_id, partition, key_str_str_len, key_str, buffer_size,
-                               cbor_buf, &err, NULL)) {
-            printf("Failed to send message config get\n");
-          } else {
-            printf("Succesfully sent config get msg\n");
+            printf("Failed to encode float message...");
           }
           break;
         }
         case STR: {
           if (cbor_encode_text_stringz(&encoder, value_str) != CborNoError) {
-            break;
-          }
-          if (!bcmp_config_set(node_id, partition, key_str_str_len, key_str, buffer_size,
-                               cbor_buf, &err, NULL)) {
-            printf("Failed to send message config get\n");
-          } else {
-            printf("Succesfully sent config get msg\n");
+            printf("Failed to encode string message...");
           }
           break;
         }
         case BYTES: {
           if (cbor_encode_byte_string(&encoder, reinterpret_cast<const uint8_t *>(value_str),
                                       strlen(value_str)) != CborNoError) {
-            break;
-          }
-          if (!bcmp_config_set(node_id, partition, key_str_str_len, key_str, buffer_size,
-                               cbor_buf, &err, NULL)) {
-            printf("Failed to send message config get\n");
-          } else {
-            printf("Succesfully sent config get msg\n");
+            printf("Failed to encode byte message...");
           }
           break;
         }
         default:
           break;
         }
+        if (!bcmp_config_set(node_id, partition, key_str_str_len, key_str, buffer_size,
+                             cbor_buf, &err, NULL)) {
+          printf("Failed to send message config set\n");
+        } else {
+          printf("Succesfully sent config set msg\n");
```

Comment:
It's all fine, but here are a couple nit picks:

- If we need both cases anyway, then the `!` in the if expression just seems confusing to me. I'd prefer to remove it and swap the success and failure clauses.
- Let's use the same phrasing in both the success and failure cases, for example, "Failed to send config set message" and "Successfully sent config set message".

Reply:
Agreed, that sounds good!